### DISCUSSION TITLE
dsp-code question

### DISCUSSION THREAD

**Discussion by coderofsalvation** [bangcorrupt/hacktribe#154] [Q&A] [answered]
First of all: this project is amazing.
Is it correct that this firmware adds certain IFX effects to es2? (like KP2 COMP)
If so, what do we know about these IFX?
what kind of dspcode/opcodes, and do IFX patches contain similar instruction-sequences (which might indicate adresses of the L/R input/output-adresses etc) ?

reason: there are some IFX which are not that interesting. If we could somehow overwrite an IFX  that would be awesome.

idea: maybe the 'off' IFX represents dsp-code too? That could be a great startingpoint from a reverse engineer perspective.

**Comment by bangcorrupt** [comment, marked as answer]
Thank you.  The IFX (and MFX) in the list are presets for a set of FX devices in the DSP firmware.  There are some FX devices that are not used in the factory presets (like the kp2 comp).  So far I can get a list of devices and their parameters with valid ranges, this will be documented in the near future when I complete FX editing via MIDI.  

The less interesting FX presets could be replaced with more interesting ones without changing the DSP code at all.  For example, phasers can be tempo synced, distortion has an EQ built in, etc...  We will also be able to map different parameters to the edit knob to change how the FX react.

I haven't dived deep into the Blackfin DSP code yet.  I have a debugger attached (thank you sponsors) and started disassembling, but there is a lot of stuff I want to finish with the CPU firmware before getting too far into it.

**Reply by coderofsalvation** [reply]
Thank you for your reply.
That makes sense, and I'm superhyped that you've been able to identify the devices and their parameters/ranges.
Ok so it's blackfin DSP code, didn't know that.
Thanks!
